File: docs/devmanual/snippets/wasm-manager/validator.js
export function execute() {
  let str = Host.inputString();
  let context = JSON.parse(str);
  let fooHeader = context.request.headers['foo'];
  if (fooHeader) {
    if (fooHeader === "bar") {
      Host.outputString(JSON.stringify({ result: true, error: null }));
    } else {
      Host.outputString(JSON.stringify({ result: false, error: { status: 401, message: `${fooHeader} is not authorized` }}));
    }
  } else {
    Host.outputString(JSON.stringify({ result: false, error: { status: 401, message: `you're not authorized` }}));
  }
  return 0;
}

File: docs/devmanual/snippets/wasm-manager/validator.rs
mod types;

use extism_pdk::*;

pub fn execute(Json(context): Json<types::WasmAccessValidatorContext>) -> FnResult<Json<types::WasmAccessValidatorResponse>> {
    match context.request.headers.get("foo") {
        Some(foo) => if foo == "bar" {
            Ok(Json(types::WasmAccessValidatorResponse { 
                result: true,
                error: None
            }))
        } else {
            Ok(Json(types::WasmAccessValidatorResponse { 
                result: false, 
                error: Some(types::WasmAccessValidatorError { 
                    message: format!("{} is not authorized", foo).to_owned(),  
                    status: 401
                })  
            }))
        },
        None => Ok(Json(types::WasmAccessValidatorResponse { 
            result: false, 
            error: Some(types::WasmAccessValidatorError { 
                message: "you're not authorized".to_owned(),  
                status: 401
            })  
        }))
    }
}